src/commands.jl

```diff
@@ -213,6 +213,35 @@ function next_line!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false
 end
 next_line!(frame::Frame, istoplevel::Bool=false) = next_line!(finish_and_return!, frame, istoplevel)
 
+is_macro_expansion(lin::LineInfoNode) = lin.method == Symbol("macro expansion")
+is_macro_expansion(::Any) = false
+function is_macro_expansion(frame)
+    src = frame.framecode.src
+    codeloc = src.codelocs[frame.pc]
+    codeloc == 0 && return false
+    linfo = src.linetable[codeloc]
+    return is_macro_expansion(linfo)
+end
+"""
+    pc = maybe_step_through_macro_expansion!(recurse, frame)
+    pc = maybe_step_through_macro_expansion!(frame)
+
+Return the current program counter of `frame` if it is not inside a macro expresion
+Otherwise, step through the statements of `frame` until the next statement is not inside
+a macro expansion.
+"""
+function maybe_step_through_macro_expansion!(@nospecialize(recurse), frame::Frame, istoplevel::Bool=false)
+    pc = frame.pc
+    is_macro_expansion(frame) || return pc
+    while pc !== nothing && !isa(pc, BreakpointRef)
```

Comment:
This is kinda similar to `next_until!` but the predicate there takes an expression, but here we need more than the expression.